app/models/ucas_matched_application.rb

```diff
@@ -33,9 +33,10 @@ def status
     if ucas_scheme?
       mapped_ucas_status
     else
-      ApplicationForm.find_by(candidate_id: @matching_data['Apply candidate ID'])
-        .application_choices.find_by(course_option: course.course_options)
-        .status
+      ApplicationChoice.find_by(
```

Comment:
You could do something like `ApplicationChoice.includes(:application_form).where(course_option: co, application_form: { candidate_id: c })`